.github/CODEOWNERS

```
@@ -0,0 +1,36 @@
+# Lines starting with '#' are comments.
+# Each line is a file pattern followed by one or more owners.
+
+# These owners will be the default owners for everything in the repo.
+# Right now there is not default owner to avoid spam
+# * @pierre-sassoulas @DanielNoord @cdce8p @jacobtylerwalls @hippo91
+
+# Order is important. The last matching pattern has the most precedence.
+
+### Core components
+
+# internal message handling
+pylint/message/*    @pierre-sassoulas
+tests/message/*    @pierre-sassoulas
+
+# typing
+pylint/typing.py     @DanielNoord @cdce8p
+
+# multiprocessing (doublethefish is not yet a contributor with write access)
+# pylint/lint/parallel.py @doublethefish
+# tests/test_check_parallel.py @doublethefish
+
+### Pyreverse
+pylint/pyreverse/*    @DudeNr33
+tests/pyreverse/*    @DudeNr33
+
+### Extensions
+
+# For any all
+pylint/extensions/for_any_all.py @areveny
```

Are these globs or regexes? Can we use `*for_any_all*`?

---

It's the same format than a .gitignore with a few exceptions : https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#syntax-exceptions

---

Yeah, so then I think **/for_any_all*/** should work!

---

```suggestion
pylint/extensions/for_any_all* @areveny
```
Do you mind if we only do this right now? I'm not sure about the syntax.

---

Why not try the desired syntax? If we can't use a full glob pattern than I think this file might quickly become unmanageable..

---

In ``**/for_any_all*/**`` Isn't the last ``/`` preventing matching the file ? Also how many files do we expect to have to match for one extension ? Maybe multiple functional tests and a directory instead of one file for the actual code, but do we really need more ?

---

> In `**/for_any_all*/**` Isn't the last `/` preventing matching the file ? 

Oh yeah. Hm, do we need 2 patterns then? Or does `glob` allow for this?

> Also how many files do we expect to have to match for one extension ? Maybe multiple functional tests and a directory instead of one file for the actual code, but do we really need more ?

Probably not, but I think we should explore whether this can be created in a somewhat maintainable state. If it just becomes a list of all files we have then I don't think we should go to the effort of creating and maintaining it.

---

> If it just becomes a list of all files we have

I don't think we should have a code owner for all files. It's rather rare that a file is complex enough to need a code owner (and it also need to be done almost entirely by a single person, that also have write right on pylint)

---

Yeah I mean you're likely the one who has to do most of the work on the file, so if you're okay with it I guess we could go ahead?